File: examples/node/custom-model-training/dataset/asticaDataset_model_list.js
const axios = require('axios');

const astica_endpoint = 'https://train.astica.ai/model/list';
const astica_api_key = '26d7e90d0b2090b30b921ls1'; //visit https://astica.ai

const astica_requestData = {
    'tkn': astica_api_key
 };

axios.post(astica_endpoint, astica_requestData, { headers: { 'Content-Type': 'application/json' } })
.then((response) => {
    const data = response.data;

    // raw output
    console.log(data);

    // handle response
    if (data.status !== 'OK') {
        console.error('API request failed');
        console.error(data.error);
    } else {
        for (let model of data['dataset_list']) {
            console.log("======================================Model:");
            console.log("title: " + model['title']);
            console.log("total_size: " + model['total_size'] + " MB");
            console.log("numClass: " + model['numClass']);
            console.log("numSample: " + model['numSample']);

            if (model['trained'] == 1) {
                console.log("trained: Yes");
                console.log("model_version: " + model['model_version']);
                console.log("date_trained: " + model['date_trained']);
            } else {
                console.log("trained: No");
            }
            console.log("date_created: " + model['date_created']);
            console.log("type: " + model['type']);
            console.log("purpose: " + model['purpose']);
            console.log("tags: " + model['tags']);
        }
    }
})
.catch((error) => {
    console.error('Unable to fetch models');
    console.error('Error:', error.toString());
});

File: examples/python-samples/custom-model-training/dataset/asticaDataset_model_list.py
import requests
import json

astica_endpoint = 'https://train.astica.ai/model/list'
astica_api_key = 'API KEY HERE' #visit https://astica.ai

astica_requestData = {
    'tkn': astica_api_key
}
try:
    response = requests.post(astica_endpoint, data=json.dumps(astica_requestData), headers={'Content-Type': 'application/json'})
    response.raise_for_status()  # Raise an exception if the request was unsuccessful
    data = json.loads(response.text)    
    #raw output
    print(data)
    #handle response
    if data['status'] != 'OK':
        print("api request failed")
        print(data["error"])
    else:
        for model in data['dataset_list']:
            print("======================================Model:") 
            print("title: "+ model['title']) 
            print("total_size: "+str(model['total_size'])+" MB") 
            print("numClass: "+str(model['numClass']))
            print("numSample: "+str(model['numSample']))
            if model['trained'] == 1:             
                print("trained: trained") 
                print("model_version: "+model['model_version']) 
                print("date_trained: "+str(model['date_trained'])) 
            else:
                print("trained: untrained") 
            print("date_created: "+str(model['date_created'])) 
            print("type: "+str(model['type'])) 
            print("purpose: "+model['purpose']) 
            print("tags: "+model['tags'])      
        
except requests.exceptions.RequestException as e:
    print("Unable to remove model")
    print("Error:", str(e))